Assets/MixedRealityToolkit.Tests/TestUtilities.cs

```diff
@@ -22,23 +21,76 @@ namespace Microsoft.MixedReality.Toolkit.Tests
 {
     public static class TestUtilities
     {
-        public static Scene testScene;
+        const string primaryTestSceneTemporarySavePath = "Assets/__temp_primary_test_scene.unity";
+        const string additiveTestSceneTemporarySavePath = "Assets/__temp_additive_test_scene_#.unity";
+        public static Scene primaryTestScene;
+        public static Scene[] additiveTestScenes = new Scene[0];
 
         public static void InitializeMixedRealityToolkit()
         {
             MixedRealityToolkit.ConfirmInitialized();
         }
 
-        public static void CleanupScene()
+        public static void TearDownScenes()
         {
-            // Create a default test scene.
+#if UNITY_EDITOR
+            // If any of our scenes were saved, tear down the assets
+            SceneAsset primaryTestSceneAsset = AssetDatabase.LoadAssetAtPath<SceneAsset>(primaryTestSceneTemporarySavePath);
+            if (primaryTestSceneAsset != null)
+            {
+                AssetDatabase.DeleteAsset(primaryTestSceneTemporarySavePath);
+            }
+
+            for (int i = 0; i < additiveTestScenes.Length; i++)
+            {
+                string path = additiveTestSceneTemporarySavePath.Replace("#", i.ToString());
+                SceneAsset additiveTestSceneAsset = AssetDatabase.LoadAssetAtPath<SceneAsset>(path);
+                if (additiveTestSceneAsset != null)
+                {
+                    AssetDatabase.DeleteAsset(path);
+                }
+            }
+            AssetDatabase.Refresh();
+#endif
+        }
+
+        public static void CreateScenes(int numScenesToCreate = 1)
+        {
+            Debug.Assert(numScenesToCreate > 0);
+
+            // Create default test scenes.
             // In the editor this can be done using EditorSceneManager with a default setup.
             // In playmode the scene needs to be set up manually.
 
 #if UNITY_EDITOR
             if (!EditorApplication.isPlaying)
             {
-                EditorSceneManager.NewScene(NewSceneSetup.DefaultGameObjects, NewSceneMode.Single);
+                List<Scene> additiveTestScenesList = new List<Scene>();
+
+                if (numScenesToCreate == 1)
+                {   // No need to save this scene, we're just creating one
+                    primaryTestScene = EditorSceneManager.NewScene(NewSceneSetup.DefaultGameObjects, NewSceneMode.Single);
+                }
+                else
+                {
+                    // Make the first scene single so it blows away previously loaded scenes
+                    primaryTestScene = EditorSceneManager.NewScene(NewSceneSetup.DefaultGameObjects, NewSceneMode.Single);
+                    // Save the scene (temporarily) so we can load additively on top of it
+                    EditorSceneManager.SaveScene(primaryTestScene, primaryTestSceneTemporarySavePath);
```

Comment:
Where do you later on unsave / remove the scene? Would this cause lots of scenes to show up saved in your assets folder?

Reply:
They get destroyed in `TestUtilities.TearDownScenes()`